lib/zendesk_apps_support/validations/svg.rb

```diff
@@ -31,27 +33,43 @@ module Svg
       end
 
       class << self
+        def contains_embedded_bitmap?(markup)
+          Nokogiri::XML(markup).xpath('//image').any?
+        end
+
+        def rewrite_svg(svg, new_markup, package, errors)
+          warning_string = if contains_embedded_bitmap?(svg.read)
+                             'txt.apps.admin.warning.app_build.bitmap_in_svg'
+                           else
+                             'txt.apps.admin.warning.app_build.sanitised_svg'
+                           end
+
+          package.warnings << I18n.t(warning_string, svg: svg.relative_path)
+          compressed_new_markup = new_markup.tr("\n", '').squeeze(' ').gsub(/\>\s+\</, '><')
```

Comment:
why do we need to parse this new_markup, isn't it coming from 'default_app_logo.svg', so it will be correct?

Reply:
I chose to store the icon uncompressed in the repo and looking like the markup that comes from Garden - I thought it was a little more accessible that way. Passing it back through here so that on the server it's stored as the stripped down/compressed version.